### PR TITLE
[CSM] Allow enabling and disabling mods.

### PR DESCRIPTION
**Pull request by red-001**
adds a new file called `mods.conf` to `/clientmods` to allow enabling/disabling mods (default:disabled) the file uses the same format for this as `world.mt`

### CODE REVIEW

**Comment by nerzhul** [comment]
This works, but we should not use configuration directly, we need to have this in main menu formspec, CSM is intended to be used by anybody

**Comment by sofar** [comment]
Actually

I'd rather prefer not to confuse the normal player with yet another menu that is empty.

An advanced user will understand the config and what client mods do, they don't really need a menu.

We really need to get servers to send clientmods first, that's more important than a formspec menu imho.

**Comment by nerzhul** [comment]
Can you factorize loading code with server part ?

**Comment by red-001** [comment]
@nerzhul done.

File: src/mods.cpp [hunk 1]
```diff
@@ -214,6 +214,55 @@ void ModConfiguration::addMods(const std::vector<ModSpec> &new_mods)
 	}
 }
 
+void ModConfiguration::addModsFormConfig(std::string settings_path, std::set<std::string>mods)
```
Comment (nerzhul) [inline src/mods.cpp]: const std::string &, const std::set &

Comment (red-001) [inline src/mods.cpp]: @nerzhul done

**Comment by rubenwardy** [comment]
I agree with sofar. I don't think this is needed at this stage, especially as client mods aren't fully released / supported. The main use of CSM should be to add client-side prediction for server mods.

**Comment by nerzhul** [comment]
Then your logic is to refuse a ready PR for a non existing PR especially for this release (feature freeze next weekend)

**Comment by red-001** [comment]
@nerzhul  @rubenwardy changed his comment

**Comment by nerzhul** [comment]
@red-001 oh i was remote, mails notifications doesn't tell us when a modification occurs

**Comment by ShadowNinja** [comment]
Untested, but looks good to me.

**Comment by paramat** [comment]
@red-001 this should have a description to make it easier to assess and review.

**Comment by Fixer-007** [comment]
Is not this already merged somehow? I already see clientmods/mods.conf in my newest engine.

**Comment by nerzhul** [comment]
it's okay, nice job

**Pull request merged**